File: src/WeddingBidders.WebApp/projects/wedding-bidders/src/app/core/models/message.ts
export interface Message {
  messageId: string;
  conversationId?: string;
  toProfileId?: string;
  fromProfileId?: string;
  subject?: string;
  content: string;
  isRead: boolean;
  createdDate: string;
}

export interface Conversation {
  conversationId: string;
  messages: Message[];
  profiles: ProfileSummary[];
}

export interface ProfileSummary {
  profileId: string;
  firstname: string;
  lastname: string;
}

export interface SendMessageRequest {
  otherProfileId: string;
  content: string;
}

File: src/WeddingBidders.Core/Model/MessageAggregate/Message.cs
namespace WeddingBidders.Core.Model.MessageAggregate;

public class Message
{
    public Guid MessageId { get; set; }
    public Guid? ConversationId { get; set; }
    public Guid? ToProfileId { get; set; }
    public Guid? FromProfileId { get; set; }
    public string? Subject { get; set; }
    public string Content { get; set; } = string.Empty;
    public bool IsRead { get; set; }
    public bool IsDeleted { get; set; }
    public DateTime CreatedDate { get; set; }
    public DateTime? LastModifiedDate { get; set; }
    public Conversation? Conversation { get; set; }
}
